be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -165,8 +165,11 @@ Status SegmentWriter::append_block(const vectorized::Block* block, size_t row_po
 }
 
 int64_t SegmentWriter::max_row_to_add(size_t row_avg_size_in_bytes) {
-    int64_t size_rows =
-            ((int64_t)MAX_SEGMENT_SIZE - (int64_t)estimate_segment_size()) / row_avg_size_in_bytes;
+    auto segment_size = estimate_segment_size();
+    if (PREDICT_FALSE(segment_size >= MAX_SEGMENT_SIZE || _row_count >= _max_row_per_segment)) {
```

Comment:
why not just use if (segment_size >= MAX_SEGMENT_SIZE || _row_count >= _max_row_per_segment) ?